docs/DEPRECATIONS.md

```diff
@@ -0,0 +1,9 @@
+# To be deprecated
+
+# To be migrated
+
+* legacy_openssl_provider v0.33.0 OpenSSL legacy provider flag should default to false
+
+# To be removed
+
+* legacy_openssl_provider v0.34.0 OpenSSL legacy provider flag should be removed
```

Comment:
The process, as described, would have us only add this once the migration above is done. That is, a deprecation would only exist in one of the three blocks at any given time.

Reply:
Ah, I see, that makes sense. Let me drop this one.